py3status/modules/hueshift.py

```diff
@@ -0,0 +1,177 @@
+```
+# good rules; show command name; don't colorize everything
+hueshift {
+    format = '[\?if=disabled&color=bad \u263c!|[\?color=hue&show \u263c '
+    format += '{command}] {temperature}K]'
+    minimum = 3000
+    maximum = 10000
+}
+```
+
+@author lasers
+
+SAMPLE OUTPUT
+# warm
+{'full_text': u'\u263c 3000K', 'color': #f3c},
+
+# neutral
+{'full_text': u'\u263c 6500K', 'color': #ff3}
+
+# cool
+{'full_text': u'\u263c 10000K', 'color': #3cf},
+
+disabled
+{'full_text': u'\u263c!', 'color': #f00}
+"""
+
+STRING_BAD_COMMAND = 'invalid command `%s`'
+STRING_NOT_INSTALLED = 'command `%s` not installed'
+STRING_NOT_AVAILABLE = 'no available command'
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    button_down = 5
+    button_toggle = 1
+    button_up = 4
+    command = None
+    delta = 250
+    format = u'\?if=disabled&color=bad \u263c!|\?color=hue \u263c {temperature}K'
+    maximum = 25000
+    minimum = 1000
+    thresholds = [(6499, '#f6c'), (6500, '#ff6'), (6501, '#6cf')]
+
+    def post_config_hook(self):
+        hueshift_commands = ['^blueshift$', '^redshift$', '^sct$']
+        check_commands = [x.strip('^$') for x in hueshift_commands]
+        self.pgrep_command = ['pgrep', '|'.join(hueshift_commands)]
+        if not self.command:
+            self.command = self.py3.check_commands(check_commands)
+        elif self.command not in check_commands:
+            raise Exception(STRING_BAD_COMMAND % self.command)
+        elif not self.py3.check_commands(self.command):
+            raise Exception(STRING_NOT_INSTALLED % self.command)
+        if not self.command:
+            raise Exception(STRING_NOT_AVAILABLE)
+
+        self.hue = {
+            'blueshift': {
+                'set': lambda value: ['blueshift', '-p', '-t', str(value)],
+            },
+            'redshift': {
+                'set': lambda value: ['redshift', '-r', '-O', str(value)],
+            },
+            'sct': {
+                'set': lambda value: ['sct', str(value)],
+            },
+        }
+
+        self.default = 6500
+        self.minimum = max(self.minimum, 1000)
+        self.maximum = min(self.maximum, 25000)
+        if self.command == 'sct':
+            self.maximum = min(self.maximum, 10000)
+
+        self.temperature = self.py3.storage_get('temperature') or self.default
+        self.last_value = self.temperature
+        self._set_color_setter_boolean()
+        if not self.is_color_setter_running:
+            self._run_command('set', self.temperature)
+
+    def _run_command(self, action=None, value=None, last=None):
+        command = self.hue[self.command][action](value)
+        self.py3.command_run(command)
+        if not last:
+            self.last_value = value
+        self.temperature = value
+
+    def _set_color_setter_boolean(self):
+        # for better user experience, we check for running processes
+        # before allowing users to shift color temperature on the screen.
```

Comment:
This comment is still poor - we check for running processes that set the screen colour

comments are only worth having if they tell the truth